### PR TITLE
framework/st_things: Fix a secure storage bug in st things

### PR DESCRIPTION
**Pull request by HONGCHAEHEE**
- Add 'root' declaration and NULL check in data manager
- Enable CONFIG_ST_THINGS_SECURESTORAGE and CONFIG_WIFI_PROFILE_SECURESTORAGE

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a181ca6aca236295b56bc541c80bbfeefded7d5a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a181ca6aca236295b56bc541c80bbfeefded7d5a] - Code Rule Check (C++) OK.

File: build/configs/artik053/st_things/defconfig [hunk 1]
```diff
@@ -811,6 +830,7 @@ CONFIG_ELOOP_POLL=y
 # CONFIG_WPA_SUPPLICANT_CMD is not set
 CONFIG_DRIVER_T20=y
 # CONFIG_ENABLE_EAP_FOR_SUPPLICANT is not set
+# CONFIG_WIFI_PROFILE_SECURESTORAGE is not set
```
Comment (JeongJunSik) [inline build/configs/artik053/st_things/defconfig]: Why did you turn off this flag?
Don't you need to encrypt WiFi information?

Comment (HONGCHAEHEE) [reply build/configs/artik053/st_things/defconfig]: Yes, this config needs to be enabled.

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 2]
```diff
@@ -616,7 +617,7 @@ char *get_json_string_from_securestorage(void)
 	}
 	cert_data[cert_size + 1] = 0;
 
-	cJSON *root = cJSON_Parse(read_data);
+	root = cJSON_Parse(read_data);
```
[the remark this answers is not in the document]
Comment (HONGCHAEHEE) [reply framework/src/st_things/things_stack/framework/things_data_manager.c]: I think so.

**Comment by seinfra** [comment]
Target : [0e1af71cc0aaf540b3f3ec26bf2f382d936a50ba] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0e1af71cc0aaf540b3f3ec26bf2f382d936a50ba] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f9eb05ee4e3c7957d8846b1b0c3950bf62b42258] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [f9eb05ee4e3c7957d8846b1b0c3950bf62b42258] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [bb44992a31916f4f762a33972ff030e906d1ce16] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [bb44992a31916f4f762a33972ff030e906d1ce16] - Code Rule Check OK.

File: build/configs/artik053s/st_things/defconfig [hunk 3]
```diff
@@ -946,18 +969,19 @@ CONFIG_DEBUG_VERBOSE=y
 #
 # Subsystem Debug Options
 #
-CONFIG_DEBUG_ST_THINGS=y
-CONFIG_DEBUG_ST_THINGS_ERROR=y
-# CONFIG_DEBUG_ST_THINGS_DEBUG is not set
-CONFIG_DEBUG_ST_THINGS_INFO=y
-# CONFIG_DEBUG_FS is not set
-# CONFIG_DEBUG_AUDIO is not set
```
[the remark this answers is not in the document]
Comment (HONGCHAEHEE) [reply build/configs/artik053s/st_things/defconfig]: I didn't change this config.

File: build/configs/artik053/st_things/defconfig [hunk 4]
```diff
@@ -246,6 +248,12 @@ CONFIG_ARTIK05X_AUTOMOUNT_SSSRW_MOUNTPOINT="/sss"
 CONFIG_ARTIK05X_AUTOMOUNT_ROMFS=y
 CONFIG_ARTIK05X_AUTOMOUNT_ROMFS_DEVNAME="/dev/mtdblock9"
 CONFIG_ARTIK05X_AUTOMOUNT_ROMFS_MOUNTPOINT="/rom"
+# CONFIG_SVR_DB_SECURESTORAGE is not set
```
[the remark this answers is not in the document]
Comment (HONGCHAEHEE) [reply build/configs/artik053/st_things/defconfig]: You're right. I set these configs to 'y'.

File: build/configs/artik053s/st_things/defconfig (hunk 3 above)
[the remark this answers is not in the document]
Comment (chanijjani) [reply build/configs/artik053s/st_things/defconfig]: Okay~.

**Pull request merged**